### ISSUE TITLE
Support maven/clj local deps pre-loading into the nix store to minimize network usage

### ISSUE THREAD

**Issue by thenonameguy** [jlesquembre/clj-nix#21]
Found this feature in https://github.com/stephank/yarn-plugin-nixify

To minimize unnecessary network requests, the Clojure source code part of the repo, especially `deps-lock` could support copying the `$HOME/.m2`, etc. folders containing the (hopefully) unedited dependencies to the `/nix/store`.

While working on a slow-ish internet connection on the project, it was quite cumbersome to test out dependency changes with 
`clj -Stree` then having `deps-lock` re-download the same exact jars/poms into the store unnecessarily.

Implementation starting at:
https://github.com/stephank/yarn-plugin-nixify/blob/321a0d7e19ee7110d0114a22aeb8e91cf549c371/src/generate.ts#L296-L300

**Comment by thenonameguy** [comment]
My 2cents:
It should be opt-out and included in the common workflow actions (`deps-lock`) implicitly.
Stretch goal would be to have it also as a separate command, that copies the _entire_ home folder contents into /nix/store with a GC root, regardless of lockfile. This minimizes the amount of network traffic used when flip-flopping between already fetched dependencies (trying to find the perfect version of `HikariCP` for instance :) ).

**Comment by jlesquembre** [comment]
> It should be opt-out and included in the common workflow actions (deps-lock) implicitly.

What about running `deps-lock` in a CI environment? For example, clojure-lsp has a github action to update the lock file: https://github.com/clojure-lsp/clojure-lsp/blob/master/.github/workflows/update_deps-lock.yml
In that case, adding the deps to the nix store makes no sense, but makes the execution longer. I didn't measure the overhead, if it's minimal, we could make it opt-out (or even don't add an opt-out option) 

> Stretch goal would be to have it also as a separate command, that copies the entire home folder contents into /nix/store with a GC root, regardless of lockfile. This minimizes the amount of network traffic used when flip-flopping between already fetched dependencies (trying to find the perfect version of HikariCP for instance :) ).

I don't see a benefit in adding the whole  `~/.m2` cache to the nix store. If the deps are added to the store with the `deps-lock` command, adding your whole cache will add only unneeded deps.

**Comment by jlesquembre** [comment]
Finally, I found some time to work on this, see #26  :)

At first, I enabled the option by default, but I noticed that it was quite slow. On my computer, generating the lock file for the [demo project](https://github.com/jlesquembre/clj-demo-project) went from ~15 seconds to ~35 seconds, so I decided to disable it by default.

The reason is that we have to invoke the `nix store add-path` command many times, which is not too fast. If at some point nix improves it, I probably will enable it by default.
